src/Aspire.Dashboard/Telemetry/AspireTelemetryService.cs

```diff
@@ -0,0 +1,364 @@
+        return new TelemetryResponse(response.StatusCode);
+    }
+
+    public async Task<ITelemetryResponse?> PostRecurringPropertyAsync(PostPropertyRequest request)
+    {
+        if (await GetHttpClientAsync().ConfigureAwait(false) is not { } client)
+        {
+            return null;
+        }
+
+        var response = await telemetrySender.MakeRequestAsync(client, c => c.PostAsJsonAsync(TelemetryEndpoints.TelemetryPostRecurringProperty, request)).ConfigureAwait(false);
+        return new TelemetryResponse(response.StatusCode);
+    }
+
+    public async Task<ITelemetryResponse?> PostCommandLineFlagsAsync(PostCommandLineFlagsRequest request)
+    {
+        if (await GetHttpClientAsync().ConfigureAwait(false) is not { } client)
+        {
+            return null;
+        }
+
+        var response = await telemetrySender.MakeRequestAsync(client, c => c.PostAsJsonAsync(TelemetryEndpoints.TelemetryPostCommandLineFlags, request)).ConfigureAwait(false);
+        return new TelemetryResponse(response.StatusCode);
+    }
+
+    public Dictionary<string, AspireTelemetryProperty> GetDefaultProperties()
+    {
+        return new Dictionary<string, AspireTelemetryProperty>
+        {
+            { TelemetryPropertyKeys.DashboardVersion, new AspireTelemetryProperty(typeof(DashboardWebApplication).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? string.Empty) },
+            { TelemetryPropertyKeys.DashboardBuildId, new AspireTelemetryProperty(typeof(DashboardWebApplication).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version ?? string.Empty) },
+        };
+    }
+
+    private async Task PerformOperationAsync(bool isUserTask, StartOperationRequest request, Func<Task<OperationResult>> func)
+    {
+        var startOperationTask = Task.Run(() => isUserTask ? StartUserTaskAsync(request) : StartOperationAsync(request));
+        var operationResult = await func().ConfigureAwait(false);
+
+        _ = Task.Run(async () =>
+        {
+            var operationId = (await startOperationTask.ConfigureAwait(false))?.Content?.OperationId;
+            if (operationId is null)
+            {
+                return;
+            }
+
+            await EndOperationAsync(new EndOperationRequest(operationId, operationResult.Result, operationResult.ErrorMessage)).ConfigureAwait(false);
+        });
+    }
+
+    private async Task<HttpClient?> GetHttpClientAsync()
+    {
+        await this.InitializeAsync().ConfigureAwait(false);
+        if (_httpClient.Value is null || _telemetrySupported is false || _telemetryEnabled is false)
+        {
+            return null;
+        }
+
+        return _httpClient.Value;
+    }
+
+    private static HttpClient? CreateHttpClient(DebugSession debugSession)
+    {
+        if (!HasDebugSession(debugSession, out var debugSessionUri, out var token, out var certData))
```

Comment:
Returning null here is weird. I think check whether there is a debug session should happen before here.

But you can still call it here and throw if code mistakenly made its way to this point.